backend/src/application/repositories/chat-room-repository.ts

```diff
@@ -0,0 +1,30 @@
+import { ChatRoom } from '@/domain';
+
+export namespace ChatRoomRepository {
+  export const Name = 'ChatRoomRepository';
+
+  export namespace Create {
+    export type Input = ChatRoom;
+
+    export type Output = ChatRoom;
+  };
+
+  export namespace GetMessages{
+    export type Input = { chatRoomId: string, };
+
+    export type Output = Pick<ChatRoom, 'messages'>;
+  }
+
+  export namespace Update{
+    export type Input = { status: 'OPEN' | 'CLOSED', };
+
+    export type Output = ChatRoom;
+  }
+
+}
+
+export interface ChatRoomRepository {
+  create: (params: ChatRoomRepository.Create.Input) => Promise<ChatRoomRepository.Create.Output>;
+  getMessages: (params: ChatRoomRepository.GetMessages.Input) => Promise<ChatRoomRepository.GetMessages.Output>;
+  update: (params: ChatRoomRepository.Update.Input) => Promise<ChatRoomRepository.Update.Output>;
```

Comment:
Precisa receber o id que vai ser editado.
```suggestion
  update: (id: string, params: ChatRoomRepository.Update.Input) => Promise<ChatRoomRepository.Update.Output>;
```